x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/AbstractUpgradeTestCase.java

```diff
@@ -30,10 +30,16 @@ public abstract class AbstractUpgradeTestCase extends ESRestTestCase {
         new SecureString(SecuritySettingsSourceField.TEST_PASSWORD)
     );
 
-    protected static final Version UPGRADE_FROM_VERSION = Version.fromString(System.getProperty("tests.upgrade_from_version"));
+    protected static final String UPGRADE_FROM_VERSION = System.getProperty("tests.upgrade_from_version");
 
     protected static final boolean SKIP_ML_TESTS = Booleans.parseBoolean(System.getProperty("tests.ml.skip", "false"));
 
+    // TODO: replace with feature testing
+    @Deprecated
+    protected static boolean isOriginalClusterVersionAtLeast(Version supportedVersion) {
```

Comment:
The purpose of this function is to not expose the old cluster version as a Version publicly, and to identify and isolate all the places were a version check is used as a feature check.

I could go a step further and encapsulate Version entirely - I have a PoC here (different class, same concepts) here: https://github.com/elastic/elasticsearch/pull/100811

Let me know if it's worth it - IMO it is, but I would not want to waste time if you have different plans or different solutions.